### PR TITLE
Add more methods to proxy

### PR DESCRIPTION
**Pull request by jamiehannaford**

### CODE REVIEW

File: rackspace/database/v1/_proxy.py [hunk 1]
```diff
@@ -280,5 +283,80 @@ def update_ha_instance(self, value, **attrs):
...
+        :param flavor_reference: The ID of the flavor which the instance will
+                                 be resized to.
+
+        :returns: None
+        """
+        instance = self._get_resource(
+            high_availability_instance.HighAvailabilityInstance, value)
+        instance.resize(self.session, flavor_reference)
+
+    def resize_volume_ha_instance(self, value, volume_size):
```
Comment (briancurtin) [inline rackspace/database/v1/_proxy.py]: What do you think about naming this `resize_ha_instance_volume` instead? Volume coming first looks slightly odd.

File: rackspace/database/v1/_proxy.py [hunk 2]
```diff
@@ -280,5 +283,80 @@ def update_ha_instance(self, value, **attrs):
...
+                                   .high_availability_instance
+                                   .HighAvailabilityInstance` instance.
+
+        :returns: None
+        """
+        instance = self._get_resource(
+            high_availability_instance.HighAvailabilityInstance, value)
+        instance.restart(self.session)
+
+    def attach_configuration_ha_instance(self, value, configuration_id):
```
Comment (briancurtin) [inline rackspace/database/v1/_proxy.py]: Can you change this one to `attach_configuration_ha_instance` to `add_configuration_to_ha_instance`? We have a precedent for this over in compute with names like `add_security_group_to_server`/`remove_security_group_from_server`.

File: rackspace/database/v1/_proxy.py [hunk 3]
```diff
@@ -280,5 +283,80 @@ def update_ha_instance(self, value, **attrs):
...
+                                   .HighAvailabilityInstance` instance.
+        :param configuration_id: The configuration ID to attach.
+
+        :returns: None
+        """
+        instance = self._get_resource(
+            high_availability_instance.HighAvailabilityInstance, value)
+        instance.attach_configuration(self.session, configuration_id)
+
+    def detach_configuration_ha_instance(self, value):
```
Comment (briancurtin) [inline rackspace/database/v1/_proxy.py]: `remove_configuration_from_ha_instance` would fit better here

File: rackspace/database/v1/high_availability_instance.py [hunk 4]
```diff
@@ -206,3 +206,27 @@ def restart(self, session):
         body = {'restart': {}}
         url = utils.urljoin(self.base_path, self.id, 'action')
         session.post(url, endpoint_filter=self.service, json=body)
+
+    def attach_configuration(self, session, configuration_id):
```
Comment (briancurtin) [inline rackspace/database/v1/high_availability_instance.py]: s/attach/add

File: rackspace/database/v1/high_availability_instance.py [hunk 5]
```diff
@@ -206,3 +206,27 @@ def restart(self, session):
...
+        :param session: The session to use for making this request.
+        :param str configuration_id: The configuration group ID.
+        :type session: :class:`~openstack.session.Session`
+        :returns: ``None``
+        """
+        body = {'ha_instance': {'configuration_id': configuration_id}}
+        url = utils.urljoin(self.base_path, self.id)
+        session.patch(url, endpoint_filter=self.service, json=body)
+
+    def detach_configuration(self, session):
```
Comment (briancurtin) [inline rackspace/database/v1/high_availability_instance.py]: s/detach/remove

**Comment by jamiehannaford** [comment]
@briancurtin Changes incorporated

**Pull request merged**